lib/AST/Decl.cpp

```diff
@@ -2980,11 +2980,32 @@ bool AbstractStorageDecl::isFormallyResilient() const {
   return true;
 }
 
+bool AbstractStorageDecl::allowBypassResilienceInPackage() const {
+  auto shouldAllow = true;
+  // Only allow bypassing resilience checks for package decls
+  // if the enclosing decl's non-resilient
+  if (auto enclosingNominal = dyn_cast<NominalTypeDecl>(getDeclContext())) {
+    shouldAllow = !enclosingNominal->isResilient();
+  } else if (auto enclosingExt = dyn_cast<ExtensionDecl>(getDeclContext())) {
+    if (auto nominal = enclosingExt->getExtendedNominal())
+      shouldAllow = !nominal->isResilient();
+  }
+  if (shouldAllow &&
+      getFormalAccessScope(/*useDC=*/nullptr,
+                           /*treatUsableFromInlineAsPublic=*/true).isPackage())
+    return getModuleContext()->allowBypassResilienceInPackage();
+  return false;
+}
+
 bool AbstractStorageDecl::isResilient() const {
   if (!isFormallyResilient())
     return false;
-
-  return getModuleContext()->isResilient();
+  if (!getModuleContext()->isResilient())
```

Comment:
What if you check isResilient() on the parent context (nominal or extension) here instead of calling isFormallyResilient()? Does anything call AbstractStorageDecl::isFormallyResilient()? We might be able to remove it entirely.

Reply:
It's called by `canStorageUseTrivialDescriptor` during SIL storage var visitor. Should it include checking package access besides the fixed layout type?

```
static bool canStorageUseTrivialDescriptor(SILGenModule &SGM,
                                           AbstractStorageDecl *decl) {
...
  if (!SGM.M.getSwiftModule()->isResilient()) {
    if (SGM.canStorageUseStoredKeyPathComponent(decl, expansion)) {
      // External modules can't directly access storage, unless this is a
      // property in a fixed-layout type.
      return !decl->isFormallyResilient();
    }
...
```